src/libraries/Native/Unix/System.Security.Cryptography.Native.Android/pal_cipher.c

```diff
@@ -178,17 +197,28 @@ int32_t AndroidCryptoNative_CipherSetKeyAndIV(CipherCtx* ctx, uint8_t* key, uint
     // CryptoNative_CipherSetKeyAndIV can be called separately for key and iv
     // so we need to wait for both and do Init after.
     if (key)
-        SaveTo(key, &ctx->key, (size_t)keyLength, /* overwrite */ true);
+        SaveTo(key, &ctx->key, (size_t)ctx->keySizeInBits, /* overwrite */ true);
```

Comment:
I think this is writing 8 times as much as it should.  (SaveTo is bytes, right?)

Reply:
Wow, sad. Fixed